de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/bpartner/service/BPartnerCreditLimitRepository.java

```diff
@@ -41,26 +51,72 @@
 @Repository
 public class BPartnerCreditLimitRepository
 {
-	public BigDecimal retrieveCreditLimitByBPartnerId(final int bpartnerId, final Timestamp date)
+	public BigDecimal getCreditLimitByBPartner(@NonNull final I_C_BPartner bpartner, @NonNull final Timestamp date)
 	{
+		final List<I_C_BPartner_CreditLimit> bpCreditLimits = retrieveCreditLimitsByBPartner(bpartner, date);
+		if (bpCreditLimits.isEmpty())
+		{
+			return BigDecimal.ZERO;
+		}
 
-		final I_C_BPartner_CreditLimit bpCreditLimit = Services.get(IQueryBL.class)
-				.createQueryBuilder(I_C_BPartner_CreditLimit.class)
-				.addEqualsFilter(I_C_BPartner_CreditLimit.COLUMNNAME_C_BPartner_ID, bpartnerId)
+		sort(bpCreditLimits);
```

Comment:
not OK, you have no guarantees that the List is mutable.
pls write this method using streams....